File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<queue<TreeNode *>> q(2);
        vector<vector<int>> v;
        q[0].push(root);
        int i = 0;
        while(1){
            if(q[0].empty() and q[1].empty())
                break;
            vector<int> w;
            while(!q[i].empty()){
                TreeNode * x = q[i].front();
                q[i].pop();
                if(!x)continue;
                w.push_back(x->val);
                q[(i+1)%2].push(x->left);
                q[(i+1)%2].push(x->right);
            }
            if(w.size())
            v.push_back(w);
            i = (i+1)%2;
        }
        return v;
        
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        q= deque()
        ans = []
        if root is None:
            return ans
        q.append(root)
        while len(q)>0:
            sz = len(q)
            t = []
            while sz > 0:
                sz-=1
                u = q.popleft()
                t.append(u.val)
                if u.left: q.append(u.left)
                if u.right: q.append(u.right)
            ans.append(t[::])
        return ans        